openpathsampling/tools.py

```diff
@@ -34,18 +34,36 @@ def in_ipynb():
 except ImportError:
     is_ipynb = False
 
+last_output = None
 
 def refresh_output(output_str, print_anyway=True, refresh=True,
                    output_stream=None, ipynb_display_only=False):
+
+    global last_output
+
     if output_stream is None:
         output_stream = sys.stdout
 
-    if is_ipynb or not ipynb_display_only or print_anyway:
-        if refresh:
+    if not output_str:
+        return
+
+    if refresh:
+        if is_ipynb:
             IPython.display.clear_output(wait=True)
+        else:
+            if last_output is not None:
+                lines = len(last_output.split('\n'))
+                CURSOR_UP_ONE = '\x1b[1A'
+                ERASE_LINE = '\x1b[2K'
+
+                output_stream.write((CURSOR_UP_ONE + ERASE_LINE) * (lines - 1))
```

Comment:
I guess we should answer for us, if we want this to mimic the behaviour of an output cell? My (still existing) problem with the approach is that if we wanted to do just that, we have no way of knowing, if we now run in a next cell and hence keep the last result or replace the last result.

Does that make sense? Example

```
# cell 1
refresh_output('step1.1')
refresh_output('step1.2')

#cell 2
refresh_output('step2.1')
```

should look in the console like
```
step 1.2
step 2.1
```

So we might put this functionality into a new function that needs a `new_cell` command to make this clear. This one could of course also work in a notebook.